samples/src/test/java/org/autorefactor/refactoring/rules/samples_in/AndroidViewHolderSample.java

```diff
@@ -0,0 +1,115 @@
+package org.autorefactor.refactoring.rules.samples_in;
+
+import android.view.LayoutInflater;
+import android.view.View;
+import android.view.ViewGroup;
+import android.widget.BaseAdapter;
+import android.widget.TextView;
+
+public abstract class AndroidViewHolderSample extends BaseAdapter {
+    @Override
+    public int getCount() {
+        return 0;
+    }
+
+    @Override
+    public Object getItem(int position) {
+        return null;
+    }
+
+    @Override
+    public long getItemId(int position) {
+        return 0;
+    }
+
+    public static class Adapter1 extends AndroidViewHolderSample {
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            return null;
+        }
+    }
+
+    public static class Adapter2 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            convertView = mInflater.inflate(R.layout.your_layout, null);
+
+            TextView text = (TextView) convertView.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return convertView;
+        }
+    }
+
+    public static class Adapter3 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        @Override
+        public View getView(int position, View convertView, ViewGroup parent) {
+            View v = mInflater.inflate(R.layout.your_layout, null);
+
+            TextView text = (TextView) v.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return v;
+        }
+    }
+
+    public static class Adapter5 extends AndroidViewHolderSample {
+        LayoutInflater mInflater;
+
+        public View getView(int position, View convertView, ViewGroup parent) {
+            // Already using View Holder pattern
+            convertView = convertView == null ? mInflater.inflate(R.layout.your_layout, null) : convertView;
+
+            TextView text = (TextView) convertView.findViewById(R.id.text);
+            text.setText("Position " + position);
+
+            return convertView;
+        }
+    }
+
+    public static class Adapter7 extends AndroidViewHolderSample {
+        LayoutInflater inflater;
+
+        @Override
+        public View getView(final int position, final View convertView, final ViewGroup parent) {
+            View rootView = convertView;
+            final int itemViewType = getItemViewType(position);
+            switch (itemViewType) {
+            case 0:
+                if (rootView != null)
+                    return rootView;
+                rootView = inflater.inflate(android.R.layout.simple_list_item_1, parent, false);
+                break;
+            }
+            return rootView;
+        }
+    }
+
+    /* TODO low priority VieHolder cornercase
+    public static class CornerCase extends ViewHolderSample {
```

Comment:
Can you convert this block comment into real code and put the expected comment as a result right after please?